src/main/java/com/google/devtools/build/lib/bazel/commands/ModCommand.java

```diff
@@ -565,14 +567,28 @@ private BlazeCommandResult runTidy(CommandEnvironment env, BazelModTidyValue mod
     }
 
     try (var stdin = CharSource.wrap(buildozerInput).asByteSource(UTF_8).openStream()) {
-      new CommandBuilder()
+      if (write) {
+        new CommandBuilder()
+            .setWorkingDir(env.getWorkspace())
+            .addArg(modTidyValue.buildozer().getPathString())
+            .addArg("-f")
+            .addArg("-")
+            .build()
+            .executeAsync(stdin, /* killSubprocessOnInterrupt= */ true)
+            .get();
+      } else {
+        var out = new CommandBuilder()
           .setWorkingDir(env.getWorkspace())
           .addArg(modTidyValue.buildozer().getPathString())
+          .addArg("-stdout")
           .addArg("-f")
           .addArg("-")
           .build()
           .executeAsync(stdin, /* killSubprocessOnInterrupt= */ true)
-          .get();
+          .get()
+          .getStdout();
+        return reportAndCreateTidyDryRunResult(env, modTidyValue, new String(out));
```

Comment:
Instead of comparing stdout for differences, which requires reading the module files, could you check `stderr` for `fixed <path>` lines? https://github.com/bazelbuild/buildtools/blob/733b4a20350d044edbef66af669b1793cf8347e3/edit/buildozer.go#L1502